clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -67,6 +67,9 @@
     /* the metadata for the client */
     private final Metadata metadata;
 
+    /* the flag indicating whether the producer should send messages in order or not. */
```

Comment:
Maybe this is a bit pedantic, but it seems to me that this is about ensuring message ordering even when there are failures and retries, not just `sending` in order. One could claim we already send in order now, but that's not enough.